addon/plugins/article-structure-plugin/commands/recalculate-structure-numbers.ts

```diff
@@ -38,30 +49,45 @@ export default function recalculateStructureNumbers(
 ) {
   const doc = transaction.doc;
   const indices = new Array<number>(structureSpecs.length).fill(1);
+
   const contexts: (PNode | null)[] = new Array<PNode | null>(
     structureSpecs.length,
   ).fill(null);
+
+  const updateNumberCalls: UpdateNumberProps[] = [];
```

Comment:
bit of a comment here would be nice, it's not the most intuitive thing (though I do think I get why you do it like this)